### PR TITLE
fix osc config to work against the same chain as osc get

### PR DESCRIPTION
**Pull request by deads2k**
This pulls in https://github.com/GoogleCloudPlatform/kubernetes/pull/6585 and wires osc config up to it.

@fabianofranz  review?
@smarterclayton This is  prerequisite to being able to fix https://github.com/openshift/origin/issues/1625 in a migrating way.

### CODE REVIEW

**Comment by smarterclayton** [comment]
How does my rebase affect this (see changes to config.go)

**Comment by deads2k** [comment]
This change supercedes that one.  I started with the factory and later realized I wanted different information.

File: Godeps/_workspace/src/github.com/GoogleCloudPlatform/kubernetes/pkg/client/clientcmd/loader.go [hunk 1]
```diff
@@ -90,8 +93,10 @@ func (rules *ClientConfigLoadingRules) Load() (*clientcmdapi.Config, error) {
 		if err := mergeConfigWithFile(mapConfig, file); err != nil {
 			errlist = append(errlist, err)
 		}
-		if err := ResolveLocalPaths(file, mapConfig); err != nil {
-			errlist = append(errlist, err)
+		if !rules.DoNotResolvePaths {
```
Comment (fabianofranz) [inline Godeps/_workspace/src/github.com/GoogleCloudPlatform/kubernetes/pkg/client/clientcmd/loader.go]: If you really want to keep the `DoNotResolvePaths` I'd suggest a `ResolvePaths()` method for clarity.

File: pkg/cmd/experimental/config/config.go [hunk 2]
```diff
@@ -3,13 +3,29 @@ package config
...
 
-	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd/config"
 	"github.com/spf13/cobra"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd/config"
+
+	cmdconfig "github.com/openshift/origin/pkg/cmd/cli/config"
 )
 
 func NewCmdConfig(parentName, name string) *cobra.Command {
```
Comment (fabianofranz) [inline pkg/cmd/experimental/config/config.go]: Is there a reason for this to be under experimental, since it's also in `osc config` already?

Comment (deads2k) [inline pkg/cmd/experimental/config/config.go]: I don't know.  @smarterclayton ?

**Comment by fabianofranz** [comment]
Make sure that stuff not in `Godeps/*` has a separate commit, not UPSTREAM.

File: Godeps/_workspace/src/github.com/GoogleCloudPlatform/kubernetes/pkg/client/clientcmd/loader.go (hunk 1 above)
Comment (deads2k) [inline Godeps/_workspace/src/github.com/GoogleCloudPlatform/kubernetes/pkg/client/clientcmd/loader.go]: Added and squashed into UPSTREAM commit.

File: pkg/cmd/experimental/config/config.go (hunk 2 above)
Comment (deads2k) [inline pkg/cmd/experimental/config/config.go]: Removed from experimental

**Comment by deads2k** [comment]
> Make sure that stuff not in Godeps/* has a separate commit, not UPSTREAM.

Split godeps into its own commit.

**Comment by deads2k** [comment]
comments addressed.

**Comment by fabianofranz** [comment]
LGTM

**Comment by deads2k** [comment]
squashed, awaiting rebase https://github.com/openshift/origin/pull/1659

**Comment by deads2k** [comment]
[merge]

**Pull request merged**